### PR TITLE
Initial API Overview document with example

### LINKED ISSUE #36: Initial API Overview document with example.
Create a initial API documentation which provides the overview of how the API will work and its usage. 

Create a example Gluster create volume API.

### PR DESCRIPTION
**Pull request by anupnivargi**
Create a initial API documentation which provides the overview of how the API will work and its usage.

Create a example Gluster create volume API.

Fixes #36

### CODE REVIEW

File: api/gluster-create-volume-example.adoc [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+Tendrl API.
+
+To begin with, we need to explore the available actions that you can take on a
+volume, to discover the actions we can use the actions API on the
+resource for the defined cluster.
+
+Sample Request
+
+----------
+GET /cluster/:cluster_id/volume/actions
```
Comment (mkudlej) [inline api/gluster-create-volume-example.adoc:14]: How do you differentiate between Gluster cluster and Ceph cluster?
Will similar Ceph functions look like this:

```
GET /cluster/:cluster_id/pool/actions
GET /cluster/:cluster_id/pool/attributes
```

I think there should be obvious that it is call about Ceph or Gluster cluster or it should be unified, for example like this:

```
GET /cluster/:cluster_id/actions
GET /cluster/:cluster_id/attributes
```

where `:cluster_id` is Ceph or Gluster cluster id.

File: api/gluster-create-volume-example.adoc [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+----------
+
+From the above response, we now know what attributes are needed to be
+submitted to the create volume API. 
+
+Sample Request
+
+----------
+POST /cluster/:cluster_id/volume/create, -d '{ volname: 'Volume 1',
+stripe_count: 1, replica_count: 3, disperse_count: 3 }'
```
Comment (shtripat) [inline api/gluster-create-volume-example.adoc:117]: This sample should contain the details of bricks as well

File: api/overview.adoc [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+`HEAD` to retrieve the HTTP header info.
+
+`GET` to retrieve a resource or multiple resources.
+
+`POST` to create a new resource.
+
+`PUT` to update a single resource or collections.
+
+`DELETE` to delete a resource
+
```
Comment (shtripat) [inline api/overview.adoc:72]: Shouldnt we have a PATCH type as well supported?

File: api/gluster-create-volume-example.adoc (hunk 1 above)
Comment (anupnivargi) [reply api/gluster-create-volume-example.adoc:14]: Yes, we know the type of the cluster with the `:cluster_id` in the URL, so we know what kind of actions and attributes are possible on that specific resource, in this case `volume`

There will also be a endpoint to find the resources which are possible on the cluster
`GET /cluster/:cluster_id/actions`

To find the attributes to create a cluster we will have
`GET /cluster/attributes`

But the example is very specific to a volume, hence it only talks about the volume actions and attributes, when we talk about the cluster creation the other endpoints will be documented.

**Review by brainfunked — COMMENTED** [review]
Looks good as a starting point.

**Pull request merged**